File: python/types/type_bool/automatic_casts_to_bool.py
#!/usr/bin/python3

i = 0

if (i): print("i = {} ({}) casted to true".format(i, type(i)))
else: print("i = {} ({}) casted to false".format(i, type(i)))

i = 1

if (i): print("i = {} ({}) casted to true".format(i, type(i)))
else: print("i = {} ({}) casted to false".format(i, type(i)))

i = 3

if (i): print("i = {} ({}) casted to true".format(i, type(i)))
else: print("i = {} ({}) casted to false".format(i, type(i)))

i = -3

if (i): print("i = {} ({}) casted to true".format(i, type(i)))
else: print("i = {} ({}) casted to false".format(i, type(i)))

print()

f = 0.0

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = 1.0

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = 3.0

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = -3.0

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = 0.1

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = 3.14

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

f = -3.14

if (i): print("f = {} ({}) casted to true".format(f, type(f)))
else: print("f = {} ({}) casted to false".format(f, type(f)))

print()

c = 0j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = 3j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = -3j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = 3+3j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = 3-3j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = -3+3j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

c = 1j

if (c): print("c = {} ({}) casted to true".format(c, type(c)))
else: print("c = {} ({}) casted to false".format(c, type(c)))

print()

s = "\0"

if (s): print("s = {} [null (\\0) character] ({}) casted to true".format(s, type(s)))
else: print("s = {} [null (\\0) character] [empty string] ({}) casted to false".format(s, type(s)))

s = ""

if (s): print("s = {} [empty string] ({}) casted to true".format(s, type(s)))
else: print("s = {} [empty string] ({}) casted to false".format(s, type(s)))

s = " "

if (s): print("s = {} [space] ({}) casted to true".format(s, type(s)))
else: print("s = {} [space] ({}) casted to false".format(s, type(s)))

s = "a"

if (s): print("s = {} ({}) casted to true".format(s, type(s)))
else: print("s = {} ({}) casted to false".format(s, type(s)))

s = "False"

if (s): print("s = {} ({}) casted to true".format(s, type(s)))
else: print("s = {} ({}) casted to false".format(s, type(s)))

s = "True"

if (s): print("s = {} ({}) casted to true".format(s, type(s)))
else: print("s = {} ({}) casted to false".format(s, type(s)))

print()

File: ruby/types/type_bool/automatic_casts_to_bool.rb
#!/usr/bin/ruby2.7

i = 0

if i then puts "i = #{i} (#{i.class}) casted to true"
else puts "i = #{i} (#{i.class}) casted to false" end

i = 1

if i then puts "i = #{i} (#{i.class}) casted to true"
else puts "i = #{i} (#{i.class}) casted to false" end

i = 3

if i then puts "i = #{i} (#{i.class}) casted to true"
else puts "i = #{i} (#{i.class}) casted to false" end

i = -3

if i then puts "i = #{i} (#{i.class}) casted to true"
else puts "i = #{i} (#{i.class}) casted to false" end

puts

f = 0.0

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = 1.0

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = 3.0

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = -3.0

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = 0.1

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = 3.14

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

f = -3.14

if f then puts "f = #{f} (#{f.class}) casted to true"
else puts "f = #{f} (#{f.class}) casted to false" end

puts

s = "\0"

if s then puts "s = #{s} [null (\\0) character] (#{s.class}) casted to true"
else puts "i = #{s} [null (\\0) character] (#{s.class}) casted to false" end

s = ""

if s then puts "s = #{s} [empty string] (#{s.class}) casted to true"
else puts "i = #{s} [empty string] (#{s.class}) casted to false" end

s = " "

if s then puts "s = #{s} [space] (#{s.class}) casted to true"
else puts "i = #{s} [space] (#{s.class}) casted to false" end

s = "a"

if s then puts "s = #{s} (#{s.class}) casted to true"
else puts "i = #{s} (#{s.class}) casted to false" end

s = "false"

if s then puts "s = #{s} (#{s.class}) casted to true"
else puts "i = #{s} (#{s.class}) casted to false" end

s = "true"

if s then puts "s = #{s} (#{s.class}) casted to true"
else puts "i = #{s} (#{s.class}) casted to false" end

puts
